### ISSUE TITLE
proxyjump functionality for remote clusters breaks after user has previously logged out of cluster or certs expired

### ISSUE THREAD

**Issue by Aharic** [gravitational/teleport#5796]
### Description

**What happened**:

If a user logs into a cluster and either manually logs out or allows certs expire, and then logs back in, it becomes impossible to run tsh proxy jump commands against a remote cluster (non-root cluster) until ~.tsh directory is removed. Setting the environment contexts and variables in 5.2 does not resolve the issue. Appears to be regression issue as the functionality worked pre 4.2 (issue discovered after upgrade).

**What you expected to happen**:

Users should not have to manually delete or clean up the local ~.tsh directory every time their certs expire or they log out when using the tsh proxy functionality. 

### Reproduction Steps
As minimally and precisely as possible, describe step-by-step how to reproduce the problem.
1. Log into root cluster.
2. Issue proxyjump command such as `tsh -J cluster2.proxy.com:3023 ssh --cluster=cluster2 --login=root -o ForwardAgent=yes cluster2_node`
3. Observe everything works as expected.
4. Logout and repeat steps 1-3, and observe an error where the same root login is not allowed.

### Server Details
- Teleport version (run `teleport version`): 4.3+

### Client Details
- Tsh version (`tsh version`): 4.3+
- Computer OS (e.g. Linux, macOS, Windows): Linux and OSX
- Installed via (e.g. apt, yum, brew, website download): official teleport binary

### Debug Logs
Please include or attach debug logs, when appropriate. Obfuscate sensitive information!

Client-Side Logs:

```▶ tsh -d -J cluster2.proxy.com:3023 ssh --cluster=cluster2 --login=root -o ForwardAgent=yes cluster2-node
INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/private/tmp/com.apple.launchd.7YuASL9Ikn/Listeners" client/api.go:2298
DEBU [KEYSTORE]  Returning SSH certificate "/Users/user/.tsh/keys/root.proxy.com/user@domain.com-cert.pub" valid until "2021-03-01 15:51:30 -0500 EST", TLS certificate "/Users/user/.tsh/keys/root.proxy.com/user@domain.co-x509.pem" valid until "2021-03-01 20:51:30 +0000 UTC". client/keystore.go:319
INFO [KEYAGENT]  Loading key for "user@domain.com" client/keyagent.go:147
DEBU [CLIENT]    Overriding SSH proxy to JumpHosts's address "cluster2.proxy.com:3023" client/api.go:1679
INFO [CLIENT]    Connecting proxy=cluster2.proxy.com:3023 login='root' method=0 client/api.go:1703
DEBU [KEYAGENT]  Validated host cluster2.proxy.com:3023. client/keyagent.go:322
INFO [CLIENT]    Successful auth with proxy cluster2.proxy.com:3023 client/api.go:1684
DEBU [CLIENT]    Found clusters: [{"name":"cluster2.proxy.com","lastconnected":"2021-03-01T20:51:06.737222448Z","status":"online"}] client/client.go:107
INFO [CLIENT]    Client= connecting to node=cluster2-node on cluster cluster2.proxy.com client/client.go:577
INFO [CLIENT]    Client= jumping to node=cluster2-node on cluster cluster2.proxy.com client/client.go:706

ERROR REPORT:
Original Error: *trace.AccessDeniedError access denied to root connecting to cluster2-node on cluster cluster2.proxy.com
Stack Trace:
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/client.go:739 github.com/gravitational/teleport/lib/client.(*ProxyClient).PortForwardToNode
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/client.go:578 github.com/gravitational/teleport/lib/client.(*ProxyClient).ConnectToNode
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/api.go:1034 github.com/gravitational/teleport/lib/client.(*TeleportClient).SSH
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:1102 main.onSSH.func1
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/lib/client/api.go:347 github.com/gravitational/teleport/lib/client.RetryWithRelogin
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:1101 main.onSSH
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:455 main.Run
	/tmp/build-darwin-amd64/go/src/github.com/gravitational/teleport/tool/tsh/tsh.go:215 main.main
	/usr/local/go/src/runtime/proc.go:213 runtime.main
	/usr/local/go/src/runtime/asm_amd64.s:1375 runtime.goexit
User Message: access denied to root connecting to cluster2-node on cluster cluster2.proxy.com```

**Comment by Aharic** [comment]
Verified that this issue does not appear with tsh client 4.1, even if cluster is on higher version.

**Comment by webvictim** [comment]
Does using a 4.1 `tsh` client function as a workaround?

**Comment by Aharic** [comment]
@webvictim It does work as a workaround for now.

**Comment by awly** [comment]
@andrejtokarcik looks related to the cluster routing work you're doing.
can you double-check and close this if it will be fixed in 6.1?

**Comment by andrejtokarcik** [comment]
Yes, this should work with the changes to be introduced in 6.1, cf. #6212. Please re-open if it turns out not to be the case, since I'm unclear on why you had to use `--cluster=cluster2` which may indicate some additional issues are involved.
